jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/DefaultHandler.java

```diff
@@ -64,42 +65,23 @@ public class DefaultHandler extends Handler.Processor
 
     public DefaultHandler()
     {
-    }
-
-    @Override
-    public void setServer(Server server)
-    {
-        super.setServer(server);
-        if (server != null)
-            initFavIcon();
-    }
-
-    private void initFavIcon()
-    {
-        if (_favicon != null)
-            return;
-
-        Server server = Objects.requireNonNull(getServer());
-
-        byte[] favbytes = null;
-        try
+        URL url = DefaultHandler.class.getResource("favicon.ico");
+        if (url == null)
         {
-            Resource faviconRes = server.getDefaultFavicon();
-            if (faviconRes != null)
-            {
-                try (InputStream is = faviconRes.newInputStream())
-                {
-                    favbytes = IO.readBytes(is);
-                }
-            }
-        }
-        catch (Exception e)
-        {
-            LOG.warn("Unable to find default favicon", e);
+            LOG.warn("Unable to find resource: %s/favicon.ico".formatted(TypeUtil.toClassReference(DefaultHandler.class.getPackageName())));
         }
-        finally
+        else
         {
-            _favicon = BufferUtil.toBuffer(favbytes);
+            try (InputStream in = url.openStream();
```

Comment:
Isn't using URL going to use the internal URL cache that will load the jar into memory again and then not use it?
Can't we use getResourceAsStream instead?